### DISCUSSION TITLE
XEP-0066 (x:oob) Support für webstorage

### DISCUSSION THREAD

**Discussion by drdownload** [SpectrumIM/spectrum2#409] [Ideas]
Conversations only displays images inline that are wrapped by an x:oob tag. http_upload from prosody does that, however images coming tru a spectrum transport with webstorage are not.

**Comment by dexterlb** [comment]
This is still a problem, and Conversations will not support xhtml. The x:oob tag is the way to go, as referenced in [this issue](https://github.com/siacs/Conversations/issues/2641).

Another possible way would be to implement a feature that uses http_upload directly instead of custom web storage. That would provide a much more streamlined way for sharing images.

**Comment by wiktor-k** [comment]
Some purple plugins like [hangouts](https://bitbucket.org/EionRobb/purple-hangouts) already send image as an URL so there is no need to upload it anywhere, "just" extract the URL from `<img src>` tag and wrap it in `x:oob` tag (if `PURPLE_MESSAGE_IMAGES` flag is set):

https://bitbucket.org/EionRobb/purple-hangouts/src/5e769791d282a6e5ae4d94dc66a1ff9d6904ec9c/hangouts_events.c?at=default&fileviewer=file-view-default#hangouts_events.c-531

**Comment by vitalyster** [comment]
should be fixed in https://github.com/SpectrumIM/spectrum2/commit/10eea7ab77a594b99482ac350a6b7f5d687e75f8
